hashes/src/main/kotlin/org/kethereum/hashes/ArrayExtensions.kt

```diff
@@ -0,0 +1,73 @@
+package org.kethereum.hashes
+
+import java.nio.ByteBuffer
+
+// INT ARRAY
+
+/**
+ * Converts the given int array into a byte array via big-endian conversion
+ * (1 int becomes 4 bytes).
+ * @return The converted array.
+ */
+fun IntArray.toByteArray(): ByteArray {
+    val array = ByteArray(this.size * 4)
+    for (i in this.indices) {
+        val bytes = this[i].toBytes()
+        array[i * 4] = bytes[0]
+        array[i * 4 + 1] = bytes[1]
+        array[i * 4 + 2] = bytes[2]
+        array[i * 4 + 3] = bytes[3]
+    }
+    return array
+}
+
+// TODO: Make this pure Kotlin
+fun IntArray.arrayCopy(srcPos: Int, dest: IntArray, destPos: Int, length: Int) {
+    System.arraycopy(this, srcPos, dest, destPos, length)
+}
+
+// BYTE ARRAY
+
+/**
+ * Converts the given byte array into an int array via big-endian conversion
+ * (4 bytes become 1 int).
+ * @return The converted array.
+ */
+fun ByteArray.toIntArray(): IntArray {
+    if (this.size % Integer.BYTES != 0) {
+        throw IllegalArgumentException("byte array length")
```

Comment:
let's state what is really the problem here like "byte array length must be a multiple of "  + Integer.BYTES